python_modules/dagster-externals/dagster_externals/_io/default.py

```diff
@@ -1,57 +1,65 @@
 import json
+from contextlib import contextmanager
+from typing import Any, Iterator, Mapping, Tuple
 
 from .._protocol import (
     DAGSTER_EXTERNALS_ENV_KEYS,
     ExternalExecutionContextData,
     ExternalExecutionMessage,
+    ExternalExecutionParams,
 )
 from .._util import DagsterExternalsError, param_from_env_var
-from .base import ExternalExecutionContextLoader, ExternalExecutionMessageWriter
+from .base import (
+    ExternalExecutionContextLoader,
+    ExternalExecutionMessageWriter,
+    ExternalExecutionParamLoader,
+)
 
 
-class ExternalExecutionFileContextLoader(ExternalExecutionContextLoader):
-    def load_context(self) -> ExternalExecutionContextData:
-        path = self._get_path_from_env()
-        with open(path, "r") as f:
-            return json.load(f)
+class PathMixin:
```

Comment:
Also definitely not convinced this `PathMixin` provides enough value over very modest code duplication, unless there is a grander plan for it

Reply:
Also generally don't like mixins where the "contract" is that a non-public property named a specific name (in this case `_path`) has to exist and it is not encoded in any other way

Reply:
I did not intend for this to be used anywhere else (should've `_`-prefixed it)-- using it as the equivalent of some private top-level functions to provide common functionality in a module.

>Also generally don't like mixins where the "contract" is that a non-public property named a specific name (in this case _path) has to exist and it is not encoded in any other way

Agree in the general case where a mixin is imported and used in remote classes but I think it's pretty clear in the tightly scoped case. That said I agree it might be harder to parse than just using a top-level private validator function. Updated.